### PR TITLE
Merge Homepage and Search

### PR DESCRIPTION
**Pull request by zbuc**
Adds homepage styling and "discover" page with filtering form, as well as `SearchController`/`SearchView` classes

### CODE REVIEW

**Comment by zbuc** [comment]
Sounds good, I backed out the CSS changes.

File: lib/queryhub_web/templates/layout/app.html.eex [hunk 1]
```diff
@@ -1,35 +1,94 @@
...
-      </main>
 
-    </div> <!-- /container -->
+          <footer class="pt-4 my-md-5 pt-md-5 border-top">
+            <div class="row">
+              <!--
+            <div class="col-12 col-md">
+              <img class="mb-2" src="" alt="QueryHub logo" width="24" height="24">
+            </div>
+            -->
```
Comment (marpaia) [inline lib/queryhub_web/templates/layout/app.html.eex]: remove or uncomment?

File: lib/queryhub_web/templates/page/index.html.eex [hunk 2]
```diff
@@ -1,36 +1,44 @@
-<div class="jumbotron">
-  <h2><%= gettext "Welcome to %{name}!", name: "Phoenix" %></h2>
-  <p class="lead">A productive web framework that<br />does not compromise speed and maintainability.</p>
+<div class="px-3 py-3 pt-md-5 pb-md-4 mx-auto text-center">
+  <h1 class="display-4"><%= gettext "%{name}", name: "Queryhub" %></h1>
```
Comment (marpaia) [inline lib/queryhub_web/templates/page/index.html.eex]: The title-case I've been going with so far is QueryHub.

File: lib/queryhub_web/templates/page/index.html.eex [hunk 3]
```diff
@@ -1,36 +1,44 @@
...
+  <p class="lead">A community application for creating and sharing <a href="https://osquery.io">osquery</a> configuration</p>
+  <p>Queryhub is an open-source web application that allows you to curate your own set of queries and packs for your osquery installation.</p>
```
Comment (marpaia) [inline lib/queryhub_web/templates/page/index.html.eex]: Same comment about QueryHub vs Queryhub.

File: lib/queryhub_web/templates/page/index.html.eex [hunk 4]
```diff
@@ -1,36 +1,44 @@
...
-    </ul>
+<div class="divider-prompt">I want to...</div>
+
+<div class="card-deck mb-3 text-center">
+  <div class="card mb-4 box-shadow">
+    <div class="card-header">
+      <h4 class="my-0 font-weight-normal">Find recommended queries</h4>
+    </div>
+    <div class="card-body">
+      <p>Queryhub features a list of recommended queries, based on curated lists from
```
Comment (marpaia) [inline lib/queryhub_web/templates/page/index.html.eex]: QueryHub

File: lib/queryhub_web/templates/page/index.html.eex [hunk 5]
```diff
@@ -1,36 +1,44 @@
...
+      contributing organizations and the most popular queries in the Queryhub community.</p>
```
Comment (marpaia) [inline lib/queryhub_web/templates/page/index.html.eex]: QueryHub

File: lib/queryhub_web/templates/page/index.html.eex [hunk 6]
```diff
@@ -1,36 +1,44 @@
...
-      <li>
-        <a href="https://twitter.com/elixirphoenix">@elixirphoenix</a>
-      </li>
-    </ul>
+  <div class="card mb-4 box-shadow">
+    <div class="card-header">
+      <h4 class="my-0 font-weight-normal">Store and Share Queries</h4>
+    </div>
+    <div class="card-body">
+      <p>You can upload, store, and share your collection of queries on Queryhub, construct new
```
Comment (marpaia) [inline lib/queryhub_web/templates/page/index.html.eex]: QueryHub

File: lib/queryhub_web/templates/layout/app.html.eex [hunk 7]
```diff
@@ -1,35 +1,94 @@
...
 <html lang="en">
-  <head>
-    <meta charset="utf-8">
-    <meta http-equiv="X-UA-Compatible" content="IE=edge">
-    <meta name="viewport" content="width=device-width, initial-scale=1">
-    <meta name="description" content="">
-    <meta name="author" content="">
 
-    <title>Hello QueryHub!</title>
-    <link rel="stylesheet" href="<%= static_path(@conn, "/css/app.css") %>">
```
Comment (marpaia) [inline lib/queryhub_web/templates/layout/app.html.eex]: I think you should still leave the header here (especially the CSS)

Comment (zbuc) [reply lib/queryhub_web/templates/layout/app.html.eex]: It's still there, line 12 in the new commit

Comment (zbuc) [reply lib/queryhub_web/templates/layout/app.html.eex]: This was just an indentation change after running an HTML formatter in VSCode

**Comment by marpaia** [comment]
screen shot 2018-06-03 at 11 50 55 pm

I thought the removal of `app.css`, but I see it's still there. I'm not sure why the styles are all out of control now though?

**Comment by zbuc** [comment]
My guess is reverting from the latest version of Bootstrap to the 3.x version packaged with Phoenix.

Re-adding the latest Bootstrap CSS should fix it, but I reverted all CSS changes on the branch

**Comment by marpaia** [comment]
The changes to `lib/queryhub_web/templates/layout/app.html.eex` are the culprit. I'm happy to close #29 if you want to update the footer links more atomically and preserve the rest of the existing layout, or we can merge that into this branch. At that point, I think this would be pretty good to go.

**Pull request merged**